src/modules/commander/Arming/PreFlightCheck/checks/powerCheck.cpp

```diff
@@ -48,36 +48,35 @@ bool PreFlightCheck::powerCheck(orb_advert_t *mavlink_log_pub, const vehicle_sta
 	if (!prearm) {
 		// Ignore power check after arming.
 		return true;
+	}
 
-	} else {
-		uORB::SubscriptionData<system_power_s> system_power_sub{ORB_ID(system_power)};
-		system_power_sub.update();
-		const system_power_s &system_power = system_power_sub.get();
+	uORB::SubscriptionData<system_power_s> system_power_sub{ORB_ID(system_power)};
+	system_power_sub.update();
+	const system_power_s &system_power = system_power_sub.get();
 
-		if (hrt_elapsed_time(&system_power.timestamp) < 200_ms) {
+	if (hrt_elapsed_time(&system_power.timestamp) < 200_ms) {
```

Comment:
Drop this 200 millisecond timeout while you're at it? 

The other thing that's a bit weird is how not having `system_power` at all results in a "pass".

Reply:
I wouldn't drop it, because otherwise you're responding to old system_power reports, which can be pretty dangerous. Its one of these "let's clean this up" reactions that re-introduces a real corner case, so let's not do that.

Also not having system power resulting in a pass is necessary for all the targets that do not have this (there are lots of them).

Reply:
I'm suggesting only running the check if updated. The problem with the timeout is that if something is misbehaving and the timestamp is badly out of date it now becomes a pass.

> Also not having system power resulting in a pass is necessary for all the targets that do not have this (there are lots of them).

Yes, I'm suggesting explicitly disabling it for those systems and then we can have a check that's actually required to pass where it matters rather than one than can fail in various ways and go unnoticed.